### PR TITLE
[new release] atdts, atds, atdpy, atdj, atdgen, atdgen-runtime, atdgen-codec-runtime and atd (2.12.0)

### PR DESCRIPTION
**Pull request by mjambon**
Project page: <a href="https://github.com/ahrefs/atd">https://github.com/ahrefs/atd</a>

##### CHANGES:

* atdgen: Annotate generated code with types to disambiguate OCaml
  classic variants (ahrefs/atd#331)
* atdpy: Support the option type more correctly so that it follows
  ATD's convention for JSON encoding. This allows compatibility with
  JSON produced by other tools of the ATD suite. The Python type,
  however, is still a nullable (`Optional`) to make things simpler for
  Python programmers. This prevents distinguishing `["Some", "None"]`
  from `"None"` which both translate to `None` in Python. (ahrefs/atd#332)
* (BREAKING) atdgen: revert default encoding of int64 values as string (ahrefs/atd#330)
* atdgen: Support `<json repr="string">` for `int` values (ahrefs/atd#330)
* atdpy: Treat default field values as expressions to evaluate each time
  they're assigned to a field. This allows the use of mutable defaults such as
  lists (ahrefs/atd#339)

### CODE REVIEW

File: packages/atdgen/atdgen.2.12.0/opam [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+bug-reports: "https://github.com/ahrefs/atd/issues"
+depends: [
+  "dune" {>= "2.8"}
+  "ocaml" {>= "4.08"}
+  "alcotest" {with-test}
+  "atd" {>= "2.11.0"}
+  "atdgen-runtime" {>= "2.1.0"}
+  "atdgen-codec-runtime" {with-test}
+  "biniou" {>= "1.0.6"}
+  "yojson" {>= "2.0.1"}
```
Comment (mseri) [inline packages/atdgen/atdgen.2.12.0/opam:80]: This lower bound may need an update:
```
#=== ERROR while compiling atdgen.2.12.0 ======================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-variants.5.1.0+trunk | pinned(https://github.com/ahrefs/atd/releases/download/2.12.0/atdts-2.12.0.tbz)
# path                 ~/.opam/5.1/.opam-switch/build/atdgen.2.12.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p atdgen -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/atdgen-7-f5ad5a.env
# output-file          ~/.opam/log/atdgen-7-f5ad5a.out
### output ###
# (cd _build/default && /home/opam/.opam/5.1/bin/ocamlopt.opt -w -40 -w -27 -safe-string -g -I atdgen/src/.atdgen_emit.objs/byte -I atdgen/src/.atdgen_emit.objs/native -I /home/opam/.opam/5.1/lib/atd -I /home/opam/.opam/5.1/lib/biniou -I /home/opam/.opam/5.1/lib/easy-format -I /home/opam/.opam/5.1/lib/re -I /home/opam/.opam/5.1/lib/yojson -intf-suffix .ml -no-alias-deps -open Atdgen_emit -o atdgen/src/.atdgen_emit.objs/native/atdgen_emit__Oj_mapping.cmx -c -impl atdgen/src/oj_mapping.ml)
# File "atdgen/src/oj_mapping.ml", line 97, characters 19-36:
# 97 |            let j = Json.get_json_int an in
#                         ^^^^^^^^^^^^^^^^^
# Error: Unbound value Json.get_json_int
# Hint: Did you mean get_json_list?
```

Comment (mjambon) [reply packages/atdgen/atdgen.2.12.0/opam:80]: oops. Thank you!

**Comment by mjambon** [comment]
I just pushed a fix for the reverse dependency checks (previous atdgen releases can't use the latest atd).

**Comment by mseri** [comment]
Looks good, thanks a lot

**Pull request merged**